app/models/package_manager/go.rb

```diff
@@ -94,8 +94,24 @@ def self.project(name)
       # that is not a valid module. https://go.dev/ref/mod#goproxy-protocol describes
       # how a $base/$module/@v/$version.mod request will return back a virtual go.mod file
       # with whatever name was passed in if there is not a go.mod file found for that version.
-      canonical_name = canonical_module_name(name)
-      search_name = canonical_name.presence || name
+      name_in_go_mod = canonical_module_name(name)
+      search_name = if name_in_go_mod.present?
+                      if name.downcase.include?(name_in_go_mod.downcase)
```

Comment:
Not sure if we need to be concerned about whitespace, and/or Go project name case sensitivity here. I think the case sensitivity isn't a big deal since it would pass this check and use the "canonical" name, but more concerned about garbage input names coming in that were previously getting resolved to a good name (via the go.mod file name) that will now get resolved back to the garbage input name that [this pr](https://github.com/librariesio/libraries.io/pull/3361) was looking to fix.

Reply:
Was GoProjectVerificationWorker previously marking any project with a not-matching-the-repo go.mod name as removed? It looks like it was to me.

How do we know there are or are not go.mod names that are a good name, but not a case-insensitive substring of the repo name?